elide-integration-tests/src/test/java/com/yahoo/elide/async/integration/tests/TableExportIT.java

```diff
@@ -0,0 +1,1441 @@
+                    attr("genre", "Science Fiction"),
+                    attr("language", "English")
+            )
+    );
+
+    private static final Resource GAME_OF_THRONES = resource(
+            type("book"),
+            attributes(
+                    attr("title", "Song of Ice and Fire"),
+                    attr("genre", "Mythology Fiction"),
+                    attr("language", "English")
+            )
+    );
+
+    private static final Resource FOR_WHOM_THE_BELL_TOLLS = resource(
+            type("book"),
+            attributes(
+                    attr("title", "For Whom the Bell Tolls"),
+                    attr("genre", "Literary Fiction"),
+                    attr("language", "English")
+            )
+    );
+
+    public TableExportIT() {
+        super(AsyncIntegrationTestApplicationResourceConfig.class, JsonApiEndpoint.class.getPackage().getName());
+
+        this.port = super.getPort();
+    }
+
+    @Override
+    protected DataStoreTestHarness createHarness() {
+        DataStoreTestHarness dataStoreTestHarness = super.createHarness();
+        return new DataStoreTestHarness() {
+                public DataStore getDataStore() {
+                    return new AsyncDelayDataStore(dataStoreTestHarness.getDataStore(), 5000);
+                }
+                public void cleanseTestData() {
+                    dataStoreTestHarness.cleanseTestData();
+                }
+        };
+    }
+    /**
+     * Creates test data for all tests.
+     */
+    @BeforeEach
+    public void init() {
+        //Create Book: Ender's Game
+        given()
+                .contentType(JSONAPI_CONTENT_TYPE)
+                .accept(JSONAPI_CONTENT_TYPE)
+                .body(
+                        datum(ENDERS_GAME).toJSON()
+                )
+                .post("/book")
+                .then()
+                .statusCode(HttpStatus.SC_CREATED);
+
+        given()
+                .contentType(JSONAPI_CONTENT_TYPE)
+                .accept(JSONAPI_CONTENT_TYPE)
+                .body(
+                        datum(GAME_OF_THRONES).toJSON()
+                )
+                .post("/book")
+                .then()
+                .statusCode(HttpStatus.SC_CREATED);
+
+        given()
+                .contentType(JSONAPI_CONTENT_TYPE)
+                .accept(JSONAPI_CONTENT_TYPE)
+                .body(
+                        datum(FOR_WHOM_THE_BELL_TOLLS).toJSON()
+                )
+                .post("/book")
+                .then()
+                .statusCode(HttpStatus.SC_CREATED);
+    }
+
+    /**
+     * Various tests for a JSONAPI query as a TableExport Request with asyncAfterSeconds value set to 0.
+     * Happy Path Test Scenario 1
+     * @throws InterruptedException InterruptedException
+     * @throws IOException IOException
+     */
+    @Test
+    public void jsonApiHappyPath1() throws InterruptedException, IOException {
+
+        AsyncDelayStoreTransaction.sleep = true;
```

Comment:
This would be significantly cleaner if sleeping (and the duration) was controlled by a request header (which the AsyncDelayStore could access via the request scope).  Can we clean that up?